### ISSUE TITLE
Update docs

### ISSUE THREAD

**Issue by davewasmer** [denali-js/denali#216]
There's been lots of (awesome) changes to Denali recently, making a lot of the docs obsolete. We should figure out what the "MVP" of docs looks like, and make sure that's ready for the 0.1.0-beta.1 release.

@acorncom given your experience in these matters, I'd love to get your thoughts :wink: - what do you think are the important topics to cover, and what can wait?

**Comment by knownasilya** [comment]
Router middleware should be covered IMO, e.g. setting up CORS.

**Comment by davewasmer** [comment]
Yep, definitely. In the meantime, in case you need it: https://github.com/denali-js/denali/blob/master/config/middleware.js#L51

**Comment by knownasilya** [comment]
Yeah, are those options documented?

**Comment by davewasmer** [comment]
Not yet ;) But hopefully the source should be fairly self explanatory in the meantime

**Comment by acorncom** [comment]
Don't have time to send in PRs for this atm, but making notes here so I can come back to it (or so other folks can help).  Given recent changes to Denali, these things need to be updated:

- [ ] http://denalijs.org/master/guides/overview/introduction/ (flat out mention Elixir, why Denali is willing to position itself against it as well as Rails?)
- [ ] http://denalijs.org/master/guides/overview/app-structure/ (mentions integration tests, which no longer exist, see below)
- [ ] the integration tests page (http://denalijs.org/master/guides/testing/integration/) needs to be re-worked to acceptance tests (and updated to use the new Ava syntax)
- [ ] unit testing info should probably be fleshed out 
- [ ] add links to the api docs on serializers on the serializers page? (http://denalijs.org/master/api/data/jsonapi-serializer/) They're entirely private classes, are we going to mark anything as public to allow sub-classing those serializers?

Things we can probably document post 0.1.0-beta:

- [ ] services
- [ ] errors
- [ ] _more thoughts here later ..._

**Comment by myared** [comment]
To echo @acorncom and as a huge fan of elixir, I'd love to see some of those points made in the introduction.
